controllers/workspace/provision/deployment.go

```diff
@@ -140,6 +140,31 @@ func DeleteWorkspaceDeployment(ctx context.Context, workspace *devworkspace.DevW
 	return true, nil
 }
 
+// ScaleDeploymentToZero scales the cluster deployment to zero
+func ScaleDeploymentToZero(ctx context.Context, workspace *devworkspace.DevWorkspace, client runtimeClient.Client) error {
+	defaultRetry := time.Millisecond * 100
+	timeout := time.Second * 60
+	err := wait.Poll(defaultRetry, timeout, func() (done bool, err error) {
+		clusterDeployment, err := getClusterDeployment(common.DeploymentName(workspace.Status.WorkspaceId), workspace.Namespace, client)
+		if err != nil {
+			return false, err
+		}
+		if clusterDeployment == nil {
+			return false, nil
+		}
+
+		replicaCount := int32(0)
+		clusterDeployment.Spec.Replicas = &replicaCount
+
+		err = client.Update(ctx, clusterDeployment)
+		if err != nil {
+			return false, err
```

Comment:
We may want to gobble a conflict exception here, if we're going with polling. Otherwise there's no cases where we retry the scale-down (apart from the case above, where no deployment exists)

Reply:
fixed by removing polling :smile: